lib/http/response-validator/integrityChecker.js

```diff
@@ -53,10 +48,10 @@ IntegrityCheckerStream.prototype._flush = function(callback) {
     this.providedChecksum.length !== CHECKSUM_BYTE_LEN ||
     this.hash.digest('base64') !== this.providedChecksum.toString('base64')
   ) {
-    this.emit('error', AWS.util.error(
+    callback( AWS.util.error(
       new Error('Response fails integrity check.'),
       { code: 'ResponseChecksumMismatch' })
-    );
+    )
```

Comment:
Call `return` either on the same line as `callback` or right under. Right now `callback` might be triggered twice: once with an error and then once without an error a couple lines below.

Reply:
Here calling callback twice won't actually write anything to stream. I will fix it though